### ISSUE TITLE
[Bug]: ERROR encountered only below iOS 17 devices "Thread 1: EXC_BAD_ACCESS (code=1, address=0x0)"

### ISSUE THREAD

**Issue by syedmurtaza108** [fluttercommunity/plus_plugins#2162]
### Platform

iOS 16

### Plugin

connectivity_plus

### Version

4.0.2

### Flutter SDK

3.13.2

### Steps to reproduce

Flutter app is crashing on my iOS 16 device, throwing the error show in logs. When I remove `connectivity_plus` plugin, it starts behave normally.

### Logs

```shell
(lldb) 2023-09-27 13:12:50.112759+0500 Runner[1388:60478] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(37)] Using the Impeller rendering backend.
<SKPaymentQueue: 0x280741450>: No observers found that respond to "paymentQueue:shouldAddStorePayment:forProduct:", will not check for purchase intents
10.12.0 - [FirebaseAnalytics][I-ACS023007] Analytics v.10.12.0 started
10.12.0 - [FirebaseAnalytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
[Firebase/Crashlytics] Version 10.12.0
10.12.0 - [FirebaseAnalytics][I-ACS025036] App Delegate Proxy is disabled
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x0000000000000000
error: memory read failed for 0x0
Target 0: (Runner) stopped.
```
```


### Flutter Doctor

```shell
✓] Flutter (Channel stable, 3.13.2, on macOS 13.5.2 22G91 darwin-arm64, locale en-GB)
    • Flutter version 3.13.2 on channel stable at /Users/syedmurtaza/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ff5b5b5fa6 (5 weeks ago), 2023-08-24 08:12:28 -0500
    • Engine revision b20183e040
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/syedmurtaza/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Users/syedmurtaza/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Users/syedmurtaza/Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.82.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.72.0

[✓] Connected device (3 available)
    • Murtaza’s iPhone (mobile) • 15b7679d6175f73480d9c20b6c7fe652f292628c • ios            • iOS 16.7 20H19
    • macOS (desktop)           • macos                                    • darwin-arm64   • macOS 13.5.2 22G91 darwin-arm64
    • Chrome (web)              • chrome                                   • web-javascript • Google Chrome 117.0.5938.92

[✓] Network resources
    • All expected network resources are available.
```
```


### Checklist before submitting a bug

- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

**Comment by Vafin84** [comment]
I have the same problem 
![image](https://github.com/fluttercommunity/plus_plugins/assets/49650802/4edffe84-6645-44f4-8dc2-16bd450bb6dc)

**Comment by TeacherXue** [comment]
try to use xcode 14

**Comment by Vafin84** [comment]
> try to use xcode 14, I also have crash when i use xcode15

Thank you, this solution worked.

**Comment by Gantushig77** [comment]
IOS 16.6 iPad pro-11-inch second gen, and iPhone 13 pro 16.7 too. Devices crash on opening an app with the package.

**Comment by ohdonpiano** [comment]
same here, build was working fine with xcode 14, once update to 15 the crash begun. Crash happens also disabling impeller with --no-enable-impeller

**Comment by victor-bao** [comment]
same problem

**Comment by Francesco-FL** [comment]
Same problem with Xcode 15 and my iPhone 6S iOS 15.7.9. If I remove connectivity_plus, it works fine.

```
Xcode build done.                                           21,1s
(lldb) 2023-09-27 21:12:31.719568+0200 Runner[3791:85896] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(37)] Using the Impeller rendering backend.
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x0000000000000000
error: memory read failed for 0x0
Target 0: (Runner) stopped.
Error launching application on iPhone.
Exited
```

**Comment by fischja** [comment]
Same issue here

**Comment by vbuberen** [comment]
It looks like Xcode 15 broke something. Could somebody validate that bumping deployment target to iOS 12 as suggested [here](https://stackoverflow.com/a/77151438/7339798) helps?

**Comment by vbuberen** [comment]
I would ask someone among people with this problem to validate the fix for the plugin. To do so, add the plugin the following way:

```
  connectivity_info_plus:
    git:
      url: git@github.com:fluttercommunity/plus_plugins.git
      path: packages/connectivity_plus/connectivity_plus
      ref: update/connectivity_plus_min_ios
```

**Comment by syedmurtaza108** [comment]
@vbuberen I added the dependency as you suggested but after running `flutter pub get`, I am getting this error
```
Git error. Command: `git clone --mirror git@github.com:fluttercommunity/plus_plugins.git /Users/syedmurtaza/.pub-cache/git/cache/plus_plugins-f7f381720bb5b068ad8f1afb2964572fe47a3ec0`
stdout: 
stderr: Cloning into bare repository '/Users/syedmurtaza/.pub-cache/git/cache/plus_plugins-f7f381720bb5b068ad8f1afb2964572fe47a3ec0'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
exit code: 128
```

**Comment by vbuberen** [comment]
@syedmurtaza108 Seems like the `git` link didn't work.
Could you try this way? 

```
connectivity_plus:
    git:
      url: https://github.com/fluttercommunity/plus_plugins.git
      path: packages/connectivity_plus/connectivity_plus
      ref: update/connectivity_plus_min_ios
```

**Comment by syedmurtaza108** [comment]
> > @syedmurtaza108 Seems like the `git` link didn't work. Could you try this way?
> > ```
> > connectivity_plus:
> >     git:
> >       url: https://github.com/fluttercommunity/plus_plugins.git
> >       path: packages/connectivity_plus/connectivity_plus
> >       ref: update/connectivity_plus_min_ios
> > ```
> 
> Not the one you asked but I tried this and I am getting this error.
> 
> ```
> Because imadTestProject depends on connectivity_plus from git which doesn't exist (Could not find git ref 'update/connectivity_plus_min_ios' (fatal: ambiguous argument 'update/connectivity_plus_min_ios': unknown revision or path not in the working tree.
> Use '--' to separate paths from revisions, like this:
> 'git <command> [<revision>...] -- [<file>...]')), version solving failed.
> Process finished with exit code 69
> ```

That is because they have merged the respective PR and therefore deleted the branch that you are referring. There is no need to verify it now as someone has already verified it. See https://github.com/fluttercommunity/plus_plugins/pull/2169#issuecomment-1742752275

**Comment by vbuberen** [comment]
Indeed, I have already merged the PR as confirmed the fix. I have just updated the snippet above to point to a commit with fix to avoid confusion.
The fix will be released as `connectivity_plus: 5.0.0` version later this week.

**Comment by ulissesferreira-totalacesso** [comment]
> Indeed, I have already merged the PR as confirmed the fix. I have just updated the snippet above to point to a commit with fix to avoid confusion. The fix will be released as `connectivity_plus: 5.0.0` version later this week.

It definitively worked for me.

**Comment by jamalbtl** [comment]
I changed the **connectivity_plus** version from **^3.0.3** to **^5.0.2** and save pubspec.yaml file and it worked for me.
Note: I am using Xcode 15
